visualization/eolearn/visualization/eoexecutor_visualization.py

```diff
@@ -86,17 +90,47 @@ def make_report(self, include_logs: bool = True):
             code_css=formatter.get_style_defs()
         )
 
-        os.makedirs(self.eoexecutor.report_folder, exist_ok=True)
+        self.eoexecutor.filesystem.makedirs(self.eoexecutor.report_folder, recreate=True)
 
-        with open(self.eoexecutor.get_report_filename(), 'w') as fout:
-            fout.write(html)
+        with self.eoexecutor.filesystem.open(self.eoexecutor.get_report_path(full_path=False), 'w') as file_handle:
+            file_handle.write(html)
 
     def _create_dependency_graph(self):
         """ Provides an image of dependency graph
         """
         dot = self.eoexecutor.workflow.dependency_graph()
         return base64.b64encode(dot.pipe()).decode()
 
+    def _get_exception_stats(self):
+        """ Creates aggregated stats about exceptions
+        """
+        formatter = HtmlFormatter()
+        lexer = pygments.lexers.get_lexer_by_name('python', stripall=True)
+
+        exception_stats = defaultdict(lambda: defaultdict(lambda: 0))
+
+        for workflow_results in self.eoexecutor.execution_results:
+            if not workflow_results.error_node_uid:
+                continue
+
+            error_node = workflow_results.stats[workflow_results.error_node_uid]
+            exception_str = pygments.highlight(
+                f'{error_node.exception.__class__.__name__}: {error_node.exception}',
+                lexer,
+                formatter
+            )
+            exception_stats[error_node.node_uid][exception_str] += 1
+
+        sorted_exception_stats = []
+        for node in self.eoexecutor.workflow.get_nodes():
+            if node.uid not in exception_stats:
+                continue
+
+            node_stats = exception_stats[node.uid]
+            sorted_exception_stats.append((node.name, node.uid, sorted(node_stats.items(), key=lambda item: -item[1])))
```

Comment:
It is not clear what this line does at first glance, a comment would be nice

Reply:
I split this part into a new method where I wrote more description into a docstring.